### PR TITLE
Fixed fatal error on null post object.

### PR DESCRIPTION
**Pull request by bogdanarizancu**
Slack thread:
- https://netzstrategen.slack.com/archives/CCGG11CSF/p1658137181718119

### CODE REVIEW

File: src/SalePercentage.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ public static function saveSalePercentage($post_id, $post) {
    *
    * @implements woocommerce_sale_flash
    */
-  public static function displaySalePercentage($output, \WP_Post $post, \WC_Product $product) {
+  public static function displaySalePercentage($output, ?\WP_Post $post, ?\WC_Product $product) {
     $salePercentage = static::getProductSalePercentage($product);
```
Comment (creative-andrew) [inline src/SalePercentage.php:134]:
```suggestion
  public static function displaySalePercentage($output, ?\WP_Post $post, ?\WC_Product $product) {
    if (!product) {
    return $output;
    }
    $salePercentage = static::getProductSalePercentage($product); 
```

File: src/SalePercentage.php [hunk 2]
```diff
@@ -130,7 +130,7 @@ public static function saveSalePercentage($post_id, $post) {
    *
    * @implements woocommerce_sale_flash
    */
```
Comment (creative-andrew) [inline src/SalePercentage.php:132]:
```suggestion
    * @param \WP_Post $post | null
   *   The WordPress Post.
   * @param \WC_Product $product | null
   *   The product object.
```

File: src/SalePercentage.php (hunk 1 above)
Comment (creative-andrew) [reply src/SalePercentage.php:134]: If we don't check product, you will trigger an error calling `static::getProductSalePercentage($product); `

**Review by creative-andrew — CHANGES_REQUESTED** [review]

**Review by creative-andrew — APPROVED** [review]

File: src/SalePercentage.php [hunk 3]
```diff
@@ -123,16 +123,18 @@ public static function saveSalePercentage($post_id, $post) {
...
    *   The WordPress Post.
-   * @param \WC_Product $product
+   * @param \WC_Product|null $product
    *   The product object.
    *
    * @implements woocommerce_sale_flash
    */
-  public static function displaySalePercentage($output, \WP_Post $post, \WC_Product $product) {
+  public static function displaySalePercentage($output, ?\WP_Post $post, ?\WC_Product $product) {
     $salePercentage = static::getProductSalePercentage($product);
```
Comment (sun) [inline src/SalePercentage.php:134]: The sale percentage shouldn't be calculated if no product is passed

**Pull request merged**